app/main.py

```diff
@@ -1,6 +1,16 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    results = {}
+
+    def inner(*args) -> Any:
+        if args in results:
+            print("Getting from cache")
+            return results[args]
+        print("Calculating new result")
+        temp_res = func(*args)
+        results[args] = temp_res
```

Comment:
Write this inline. And use if-else statement(return after it)

Reply:
Sorry, i don't understand. What should i write and how? 
If i write         
 ```
    temp_res = func(*args)
    results[args] = temp_res
```
        before if statement i will can't check args in results. 
        Also i use return for finishing the function, so i don't need else statement.

Do you mean that i should write 
```
    results[args] = func(*args)
    return results[args]
```
?

Reply:
Yes, I think, that you should write this something like this.
```suggestion
        results[args] = func(*args)
```